clang-tools-extra/constexpr-everything/main.cpp

```diff
@@ -229,7 +229,7 @@ class ConstexprFunctionASTVisitor
 
     // Mark function as constexpr, the next ast visitor will use this
     // information to find constexpr vardecls
-    func->setConstexpr(true);
+    func->setConstexprKind(CSK_constexpr);
```

Comment:
why is this changed needed from the mainline?

Reply:
The mainline is aimed at clang 8 so it required this api change.